File: nodejs/database_request.js
const http = require('http');
const mysql = require('mysql');

// Konfiguracja połączenia z bazą danych
const connection = mysql.createConnection({
    host: 'localhost',
    user: 'yourUsername',
    password: 'yourPassword',
    database: 'yourDatabase'
});

connection.connect();

// Tworzenie serwera HTTP
http.createServer((req, res) => {
    // Wykonanie zapytania SELECT
    connection.query('SELECT * FROM yourTable', (error, results, fields) => {
        if (error) {
            res.writeHead(500, {'Content-Type': 'text/plain'});
            res.end('Server error');
            return;
        }

        res.writeHead(200, {'Content-Type': 'application/json'});
        res.end(JSON.stringify(results));
    });
}).listen(3000, () => {
    console.log('Server running at http://localhost:3000/');
});

File: python/database_request.py
import tornado.ioloop
import tornado.web
import mysql.connector

# Konfiguracja połączenia z bazą danych
db_connection = mysql.connector.connect(
    host='localhost',
    user='yourUsername',
    password='yourPassword',
    database='yourDatabase'
)

# Klasa obsługująca zapytanie HTTP
class MainHandler(tornado.web.RequestHandler):
    def get(self):
        try:
            # Tworzenie kursora do bazy danych
            cursor = db_connection.cursor()

            # Wykonanie zapytania SELECT
            cursor.execute('SELECT * FROM yourTable')

            # Pobranie wyników zapytania
            results = cursor.fetchall()

            # Zamknięcie kursora
            cursor.close()

            # Wysłanie wyników jako odpowiedź HTTP
            self.write({'data': results})
        except Exception as e:
            self.set_status(500)
            self.write({'error': 'Server error'})

def make_app():
    return tornado.web.Application([
        (r'/', MainHandler),
    ])

if __name__ == '__main__':
    app = make_app()
    app.listen(3000)
    print('Server running at http://localhost:3000/')
    tornado.ioloop.IOLoop.current().start()
